src/oscar/apps/offer/abstract_models.py

```diff
@@ -452,7 +452,7 @@ def products(self):
 
         queryset = self.condition.range.all_products()
         return queryset.filter(is_discountable=True).exclude(
```

Comment:
En then you can remove the .exclude since this is also done in browsable :)

Reply:
https://github.com/django-oscar/django-oscar/blob/master/src/oscar/apps/catalogue/managers.py#L108

see parent=None is the same as .exclude(structure=CHILD)